### PR TITLE
C1 kareemalirahman c1s24 25 27 vote form

### PR DESCRIPTION
**Pull request by krmali**
@adamggg implementation reviewer
@m0hamed scenario reviewer
@zoheiry testing and documentation reviewer
@Khloud-Khalid UML reviewer
issues #176 #170 #169

### CODE REVIEW

File: arability/app/models/gamer.rb [hunk 1]
```diff
@@ -189,12 +194,14 @@ def select_synonym(synonym_id)
...
+  #   keyword_id: is the id of keyword for which the synonym is being added
+  #   is_formal: determine whether the synonym is formal or slang
   # Success:
   #   returns 0 returned by the invoked method meaning saved new synonym
   # Failure:
   #   returns 1,2,3 according to the invoked method failure scenario
-  def suggest_synonym(synonym_name, keyword_id)
-    return Synonym.record_suggested_synonym(synonym_name, keyword_id)
+  def suggest_synonym(synonym_name, keyword_id, is_formal)
+    return Synonym.record_suggested_synonym(synonym_name, keyword_id, true ,is_formal)
```
Comment (adamggg) [inline arability/app/models/gamer.rb]: Remove the space after the true and add one before the is_formal

File: arability/app/models/synonym.rb [hunk 2]
```diff
@@ -35,16 +35,22 @@ class << self
   # Failure:
   #   returns 1 when the synonym written by the gamer is blank
   #   returns 2 when the synonym is already existing
-  def self.record_suggested_synonym(synonym_name, keyword_id, approved= false)
+  #   returns 4 when the gamer didn't choose formal or slang to the synonym
+  def self.record_suggested_synonym(synonym_name, keyword_id, approved= true, is_formal)
```
Comment (adamggg) [inline arability/app/models/synonym.rb]: approved = instead of approved= =)

File: arability/app/views/games/_vote.html.erb [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			</td>		
+		</tr>
+	</table>
+	<table class="vote_table">
+		<% j = 0 %>
+		<% while syn_list.size>0  && j<syn_list.length %>
+		<tr>
+			<td style="width:350px; text-align:center;">
+				<% if syn_list[j].is_formal == nil %>
+					<button data-trigger="manual" data-content="<%= t(:formal_or_slang) %>" data-placement="bottom" class="btn btn-success" id="synonym_button<%=j%>" data-html="true" type="submit" onclick="check_formality('<%= syn_list[j].id%>', '<%= syn_list[j].is_formal %>','<%= j %>')"><%= syn_list[j].name %></button>
```
Comment (adamggg) [inline arability/app/views/games/_vote.html.erb]: Add space before '<%= j %>'

File: arability/app/views/games/_vote.html.erb [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+				<% else %>
+					<button class="btn btn-success" id="synonym_button" type="submit" onclick="record_vote('<%= syn_list[j].id%>','nil')"><%= syn_list[j].name %></button>
```
Comment (adamggg) [inline arability/app/views/games/_vote.html.erb]: space before 'nil' =)

File: arability/app/views/games/_vote.html.erb [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+				<% end %>
+				<% j = j+1 %>
+			</td>
+			<td style="width:350px; text-align:center;">
+				<% if j<syn_list.length %>
```
Comment (adamggg) [inline arability/app/views/games/_vote.html.erb]: Add a space after j and after <

File: arability/app/views/games/_vote.html.erb [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+					<% if syn_list[j].is_formal == nil %>
+						<button data-trigger="manual" data-content="<%= t(:formal_or_slang) %>" data-placement="bottom" class="btn btn-success" id="synonym_button<%=j%>" data-html="true" type="submit" onclick="check_formality('<%= syn_list[j].id%>', '<%= syn_list[j].is_formal %>','<%= j %>')"><%= syn_list[j].name %></button>
+					<% else %>
+						<button class="btn btn-success" id="synonym_button" type="submit" onclick="record_vote('<%= syn_list[j].id%>','nil')"><%= syn_list[j].name %></button>
```
Comment (adamggg) [inline arability/app/views/games/_vote.html.erb]: space before 'nil' =)

File: arability/app/views/games/_vote.html.erb [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+			<td style="width:230px;"></td>
+			<td style="width:230px; text-align:center;">
+				<button class="btn btn-primary"  id="skip_button" type="submit" onclick="skip_keyword()"><%= t(:skip_this_keyword) %></button>
+			</td>
+		</tr>
+	</table>
+</div>
+<input id="choosen_synonym_id" type="hidden" value="old_value"/>
+<script type="text/javascript">
+	function check_formality(syn_id, syn_is_formal,j){
```
Comment (adamggg) [inline arability/app/views/games/_vote.html.erb]: Add space before the j

File: arability/app/views/games/_vote.html.erb [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+	}
+	function hide(){
+		$('#synonym_button0').popover('destroy');
+		$('#synonym_button1').popover('destroy');
+		$('#synonym_button2').popover('destroy');
+		$('#synonym_button3').popover('destroy');
+	}
+	function choose_formal_and_vote(){
+		var synonym_id = $("#choosen_synonym_id").val();
+		record_vote(synonym_id,"formal");
```
Comment (adamggg) [inline arability/app/views/games/_vote.html.erb]: Space before "formal"

File: arability/app/views/games/record_synonym.js.erb [hunk 9]
```diff
@@ -1,10 +1,10 @@
-<% if @record_output == 2 %>
+<% if @record_output == 2 || @record_output == 0 %>
 	<% if @already_existing_synonym %>
-		record_vote(<%= @already_existing_synonym.id %>);
-	<% end %>
-<% elsif @record_output == 0 %>
-	<% if @already_existing_synonym %>
-		record_vote(<%= @already_existing_synonym.id %>);
+		<% if @is_formal == "formal" %>
+			record_vote(<%= @already_existing_synonym.id %>,true);
```
Comment (adamggg) [inline arability/app/views/games/record_synonym.js.erb]: Space before true

File: arability/app/views/games/record_synonym.js.erb [hunk 10]
```diff
@@ -1,10 +1,10 @@
...
+		<% elsif @is_formal == "slang" %>
+			record_vote(<%= @already_existing_synonym.id %>,false);
```
Comment (adamggg) [inline arability/app/views/games/record_synonym.js.erb]: Space before false

File: arability/app/views/games/vote.js.erb [hunk 11]
```diff
@@ -1,9 +1,8 @@
 var words=get_words();
 var translation="";
 
-
-function record_vote(syn_id){
-	$.post("/games/record_vote", {synonym_id: syn_id } );
+function record_vote(syn_id, is_formal){
+	$.post("/games/record_vote", {synonym_id: syn_id, is_formal:is_formal } );
```
Comment (adamggg) [inline arability/app/views/games/vote.js.erb]: Space before is_formal

**Comment by adamggg** [comment]
@KareemAliRahman check comments in files changed please =)

File: arability/app/models/gamer.rb [hunk 12]
```diff
@@ -189,12 +194,14 @@ def select_synonym(synonym_id)
...
+  #   keyword_id: is the id of keyword for which the synonym is being added
+  #   is_formal: determine whether the synonym is formal or slang
   # Success:
   #   returns 0 returned by the invoked method meaning saved new synonym
   # Failure:
   #   returns 1,2,3 according to the invoked method failure scenario
-  def suggest_synonym(synonym_name, keyword_id)
-    return Synonym.record_suggested_synonym(synonym_name, keyword_id)
+  def suggest_synonym(synonym_name, keyword_id, is_formal)
+    return Synonym.record_suggested_synonym(synonym_name, keyword_id, true , is_formal)
```
Comment (adamggg) [inline arability/app/models/gamer.rb]: Fix this extra white space :alien:

**Comment by adamggg** [comment]
@KareemAliRahman check files changed comments =)

**Comment by Khloud-Khalid** [comment]
fix merge conflicts =)

**Pull request merged**